src/components/Card.js

```diff
@@ -1,30 +1,50 @@
 export default class Card {
-  constructor(data, cardSelector, handleCardClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(
+    { name, link, isLiked, _id, owner },
+    cardSelector,
+    handleCardClick,
+    handleDeleteBtnClick,
+    handleCardLike
+  ) {
+    this._name = name;
+    this._link = link;
+    this.isLiked = isLiked;
+    this._cardId = _id;
+    this._owner = owner._id;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteBtnClick = handleDeleteBtnClick;
+    this._handleCardLike = handleCardLike;
   }
 
   _setEventListeners() {
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeBtn();
+      this._handleCardLike(this._cardId);
     });
 
     this._deleteButton.addEventListener("click", () => {
-      this._handleDeleteBtn();
+      this._handleDeleteBtnClick(this._cardId);
     });
 
     this._cardImageElement.addEventListener("click", () => {
       this._handleCardClick({ name: this._name, link: this._link });
     });
   }
 
-  _handleLikeBtn() {
-    this._likeButton.classList.toggle("card__like-button_active");
+  toggleLikes(isLiked) {
+    this.isLiked = isLiked;
+    this._renderLikes();
```

Comment:
The logic for rendering likes is brilliant

Reply:
Thank you!